### ISSUE TITLE
Are there plans for support of FHIR custom resource JSON deserialization? 

### ISSUE THREAD

**Issue by Jmacek** [jamesagnew/hapi-fhir#1099]
Issue: When deserializing a JSON I had serialized from a custom FHIR resource, I seem to lose some of the `@Child` elements.

References and extensions work fine, but just having a child element that is an existing Fhir object, such as `DateTimeType` inside of a custom class, say one that extends Basic, loses child elements when deserializing.

Example
```
@ResourceDef(name = "Basic")
Public class MyCustomClass extends Basic
   @Child(type = DateTimeType.class)
    private DateTimeType myDateTime;

   @Child(type Reference.class)
   private Reference myReference;
```

Serializes fine, but when you deserialize, it is missing myDateTime even though it was in the serialized JSON.

This seems like a limitation of the library, as the deserialization is based around the FHIR extension `Basic`, which doesn't contain `DateTimeType myDateTime`, but are there plans to implement support for custom Resources in the future?

TL;DR: Since JSON Deserialization of custom resources are based on their extensions you lose any elements that aren't also a part of the extended object. At least I assume that is what is happening

**Comment by Jmacek** [comment]
I misread the API, there is an existing parseResource for custom resources. No issues deserializing when using that.
